File: func_log/zadania1_5/zadanie3.py
#нефункциональный 

def histogram_imperative(numbers):
    lines = []                        # изменяемое состояние
    for n in numbers:                 # явный цикл
        bar = ""
        for _ in range(n):            # ещё один вложенный цикл
            bar += "#"                # мутация строки-аккумулятора
        lines.append(bar)             # мутация списка
    return lines


if __name__ == "__main__":
    data = input("Введите числа через пробел: ")
    numbers = [int(x) for x in data.split()]
    for line in histogram_imperative(numbers):
        print(line)


#функциональный

def histogram_functional(numbers):
    return ["#" * n for n in numbers]


if __name__ == "__main__":
    data = input("Введите числа через пробел: ")
    numbers = [int(x) for x in data.split()]
    print("\n".join(histogram_functional(numbers)))

File: func_log/zadania1_5/zadanie3.js
// нефункциональный

function histogramImperative(numbers) {
    const lines = [];                 // изменяемое состояние
    for (let i = 0; i < numbers.length; i++) {   // явный счётчик
        let bar = "";
        for (let j = 0; j < numbers[i]; j++) {    // вложенный цикл
            bar += "#";               // мутация строки-аккумулятора
        }
        lines.push(bar);              // мутация массива
    }
    return lines;
}

const numbers = [10, 15, 7, 9, 1, 3];
histogramImperative(numbers).forEach(line => console.log(line));


// функциональный
const histogramFunctional = numbers =>
    numbers.map(n => "#".repeat(n));

const numbers = [10, 15, 7, 9, 1, 3];
console.log(histogramFunctional(numbers).join("\n"));